lib/bugsnag/utility/circular_buffer.rb

```diff
@@ -2,30 +2,57 @@ module Bugsnag::Utility
   class CircularBuffer
     include Enumerable
 
+    # @return [Numeric] the current maximum allowable number of items
     attr_reader :max_items
 
+    ##
+    # Creates a circular buffer
+    #
+    # @api private
+    #
+    # @param max_items [Integer] the initial maximum number of items
     def initialize(max_items = 25)
       @max_items = max_items
       @buffer = []
     end
 
+    ##
+    # Adds an item to the circular buffer
+    #
+    # If this causes the buffer to exceed its maximum items, the oldest item will be removed
+    #
+    # @param item [Any] the item to add to the buffer
```

Comment:
This should probably be `[Object]` (although I could be wrong and wouldn't be surprised if I wasn't aware of some `[Any]` or `[any]` convention in yardoc)